### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by ladanazita**
E2E test:

```
    [[SEGAnalytics sharedAnalytics] track:@"Testing if malformed"];
    [[SEGAnalytics sharedAnalytics] identify:@"3942084234230" traits:@{
                                                                       @"gender" : @"female",
                                                                       @"company" : @"segment",
                                                                       @"name" : @"ladan"
                                                                       }];
```

screenshot 2017-10-09 13 06 33

### CODE REVIEW

**Comment by ladanazita** [comment]
Travis failing at
`$ pod lib lint --use-libraries --allow-warnings`

Works locally, builds locally - removing from `.travis.yml`

File: Example/Tests/Tests.m [hunk 1]
```diff
@@ -10,39 +10,211 @@
...
+        });
     });
     
-    it(@"will wait and succeed", ^{
-        waitUntil(^(DoneCallback done) {
-            done();
+    describe(@"identify", ^{
+        it(@"identifies an unknown user with traits", ^{
+            SEGIdentifyPayload *identifyPayload = [[SEGIdentifyPayload alloc] initWithUserId:nil anonymousId:@"324908523402" traits:@{
+                                                                                                                                      @"gender" : @"female",
```
Comment (tonyxiao) [inline Example/Tests/Tests.m]: formatting seems really inconsistent, are you using space commander for all?

File: Segment-Intercom/Classes/SEGIntercomIntegration.m [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        self.intercom = [Intercom class];
+        
+        NSString *apiKey = settings[@"apiKey"];
+        NSString *iOSAppId = settings[@"iOSAppId"];
+        
+        [self.intercom setApiKey:apiKey forAppId:iOSAppId];
+        SEGLog(@"[self.intercom setApiKey:%@ forAppId:%@];",apiKey, iOSAppId);
+        
+        // For testing
+        [self.intercom enableLogging];
```
Comment (tonyxiao) [inline Segment-Intercom/Classes/SEGIntercomIntegration.m]: What if it's not testing situation?

File: Segment-Intercom/Classes/SEGIntercomIntegration.m [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    }
+    
+    NSDictionary *integration = [payload.integrations valueForKey:@"intercom"];
+    if(integration[@"languageOverride"]) {
+        userAttributes.languageOverride = integration[@"languageOverride"];
+    }
+    
+    // Intercom requires each value must be of type NSString, NSNumber or NSNull.
+    for (NSString *key in traits) {
+        if (![[traits valueForKey:key] isKindOfClass:[NSString class]] &&
```
Comment (tonyxiao) [inline Segment-Intercom/Classes/SEGIntercomIntegration.m]: what about the handling of `NSNull`?

File: Segment-Intercom/Classes/SEGIntercomIntegration.m [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    };
+    
+    if (companyTraits[@"plan"]) {
+        company.plan = companyTraits[@"plan"];
+        [customAttributes removeObjectForKey:@"plan"];
+    };
+    
+    // Intercom requires each value must be of type NSString, NSNumber or NSNull.
+    for (NSString *key in companyTraits) {
+        if (![[companyTraits valueForKey:key] isKindOfClass:[NSString class]] &&
```
Comment (tonyxiao) [inline Segment-Intercom/Classes/SEGIntercomIntegration.m]: `NSNull` is not handled here.

**Review by tonyxiao — CHANGES_REQUESTED** [review]

File: Example/Tests/Tests.m (hunk 1 above)
Comment (ladanazita) [reply Example/Tests/Tests.m]: yeah - I can re-run and see whats up

File: Segment-Intercom/Classes/SEGIntercomIntegration.h [hunk 5]
```diff
@@ -14,10 +15,10 @@
 @interface SEGIntercomIntegration : NSObject <SEGIntegration>
 
 @property (nonatomic, strong, nonnull) NSDictionary *settings;
-@property (nonatomic, strong) Class intercom;
+@property (nonatomic, strong) Class _Nullable intercom;
 
-- (instancetype)initWithSettings:(NSDictionary *_Nonnull)settings;
-- (instancetype)initWithSettings:(NSDictionary *)settings andIntercom:(id)intercom;
+- (instancetype _Nullable )initWithSettings:(NSDictionary *_Nonnull)settings;
```
Comment (tonyxiao) [inline Segment-Intercom/Classes/SEGIntercomIntegration.h]: This is actually `_Nullable`?

File: Segment-Intercom/Classes/SEGIntercomIntegration.m [hunk 6]
```diff
@@ -21,14 +21,11 @@ - (instancetype)initWithSettings:(NSDictionary *)settings
         self.settings = settings;
         self.intercom = [Intercom class];
         
-        NSString *apiKey = settings[@"apiKey"];
-        NSString *iOSAppId = settings[@"iOSAppId"];
+        NSString *mobileApiKey = settings[@"mobileApiKey"];
+        NSString *appId = settings[@"appId"];
```
Comment (tonyxiao) [inline Segment-Intercom/Classes/SEGIntercomIntegration.m]: Confirming this change is actually intended?

Comment (tonyxiao) [reply Segment-Intercom/Classes/SEGIntercomIntegration.m]: As in, did server side settings change?

**Review by tonyxiao — CHANGES_REQUESTED** [review]

File: Segment-Intercom/Classes/SEGIntercomIntegration.m (hunk 6 above)
Comment (ladanazita) [reply Segment-Intercom/Classes/SEGIntercomIntegration.m]: I had to change this because we can't expose (make public) the `apiKey` setting for server-side

**Review by tonyxiao — APPROVED** [review]

**Pull request merged**